commit 752272f16dd18f2cac58a583a8673c8e2fb93abb
Merge: 3f3c8be973af 96710247298d
Author: Linus Torvalds <torvalds@linux-foundation.org>
Date:   Mon Nov 25 18:02:36 2019 -0800

    Merge tag 'for-linus' of git://git.kernel.org/pub/scm/virt/kvm/kvm
    
    Pull KVM updates from Paolo Bonzini:
     "ARM:
       - data abort report and injection
       - steal time support
       - GICv4 performance improvements
       - vgic ITS emulation fixes
       - simplify FWB handling
       - enable halt polling counters
       - make the emulated timer PREEMPT_RT compliant
    
      s390:
       - small fixes and cleanups
       - selftest improvements
       - yield improvements
    
      PPC:
       - add capability to tell userspace whether we can single-step the
         guest
       - improve the allocation of XIVE virtual processor IDs
       - rewrite interrupt synthesis code to deliver interrupts in virtual
         mode when appropriate.
       - minor cleanups and improvements.
    
      x86:
       - XSAVES support for AMD
       - more accurate report of nested guest TSC to the nested hypervisor
       - retpoline optimizations
       - support for nested 5-level page tables
       - PMU virtualization optimizations, and improved support for nested
         PMU virtualization
       - correct latching of INITs for nested virtualization
       - IOAPIC optimization
       - TSX_CTRL virtualization for more TAA happiness
       - improved allocation and flushing of SEV ASIDs
       - many bugfixes and cleanups"
    
    * tag 'for-linus' of git://git.kernel.org/pub/scm/virt/kvm/kvm: (127 commits)
      kvm: nVMX: Relax guest IA32_FEATURE_CONTROL constraints
      KVM: x86: Grab KVM's srcu lock when setting nested state
      KVM: x86: Open code shared_msr_update() in its only caller
      KVM: Fix jump label out_free_* in kvm_init()
      KVM: x86: Remove a spurious export of a static function
      KVM: x86: create mmu/ subdirectory
      KVM: nVMX: Remove unnecessary TLB flushes on L1<->L2 switches when L1 use apic-access-page
      KVM: x86: remove set but not used variable 'called'
      KVM: nVMX: Do not mark vmcs02->apic_access_page as dirty when unpinning
      KVM: vmx: use MSR_IA32_TSX_CTRL to hard-disable TSX on guest that lack it
      KVM: vmx: implement MSR_IA32_TSX_CTRL disable RTM functionality
      KVM: x86: implement MSR_IA32_TSX_CTRL effect on CPUID
      KVM: x86: do not modify masked bits of shared MSRs
      KVM: x86: fix presentation of TSX feature in ARCH_CAPABILITIES
      KVM: PPC: Book3S HV: XIVE: Fix potential page leak on error path
      KVM: PPC: Book3S HV: XIVE: Free previous EQ page when setting up a new one
      KVM: nVMX: Assume TLB entries of L1 and L2 are tagged differently if L0 use EPT
      KVM: x86: Unexport kvm_vcpu_reload_apic_access_page()
      KVM: nVMX: add CR4_LA57 bit to nested CR4_FIXED1
      KVM: nVMX: Use semi-colon instead of comma for exit-handlers initialization
      ...

diff --cc include/Kbuild
index 95508049ee51,e8154f8bcac5..25edc43483e0
--- a/include/Kbuild
+++ b/include/Kbuild
@@@ -65,7 -65,10 +65,9 @@@ header-test-			+= keys/asymmetric-subty
  header-test-			+= keys/asymmetric-type.h
  header-test-			+= keys/big_key-type.h
  header-test-			+= keys/request_key_auth-type.h
 -header-test-			+= keys/trusted.h
  header-test-			+= kvm/arm_arch_timer.h
+ header-test-$(CONFIG_ARM)	+= kvm/arm_hypercalls.h
+ header-test-$(CONFIG_ARM64)	+= kvm/arm_hypercalls.h
  header-test-			+= kvm/arm_pmu.h
  header-test-$(CONFIG_ARM)	+= kvm/arm_psci.h
  header-test-$(CONFIG_ARM64)	+= kvm/arm_psci.h